File: 14. Longest Common Prefix/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int minLength = Integer.MAX_VALUE;
        for (String s : strs) {
            if (s.length() < minLength) {
                minLength = s.length();
            }
        }

        for (int i = 0; i < minLength; i++) {
            for (String s : strs) {
                if (strs[0].charAt(i) != s.charAt(i)) {
                    return s.substring(0, i);
                }
            }
        }

        return strs[0].substring(0, minLength);
    }
}

File: 14. Longest Common Prefix/14. Longest Common Prefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        min_length = float("inf")
        for s in strs:
            if len(s) < min_length:
                min_length = len(s)

        for i in range(min_length):
            for s in strs:
                if strs[0][i] != s[i]:
                    return s[:i]

        return strs[0][:min_length]
